diff --git a/src/Makefile b/src/Makefile
index cdcb94a..6abd8e0 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -40,8 +40,12 @@ install: module-install
 
 rwildcard=$(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2) $(filter $(subst *,%,$2),$d))
 DKMS_SOURCES := version.h Makefile Kbuild Kconfig dkms.conf $(filter-out version.h wireguard.mod.c tests/%,$(call rwildcard,,*.c *.h *.S *.pl *.include))
+DKMS_PATCHES := patches/blake2s.patch
+
 dkms-install: $(DKMS_SOURCES)
 	@$(foreach f,$(DKMS_SOURCES),install -v -m0644 -D $(f) $(DESTDIR)$(DKMSDIR)/$(f);)
+	@$(foreach f,$(DKMS_PATCHES),install -v -m0644 -D $(f) $(DESTDIR)$(DKMSDIR)/$(f);)
+	install -v -m0755 -D pre-build.sh $(DESTDIR)$(DKMSDIR)/pre-build.sh
 
 style:
 	$(KERNELDIR)/scripts/checkpatch.pl -f --max-line-length=4000 --codespell --color=always $(filter-out wireguard.mod.c,$(wildcard *.c)) $(wildcard *.h) $(wildcard selftest/*.c)
